### DISCUSSION TITLE
v3.0.0.1: Unable to start

### DISCUSSION THREAD

**Discussion by phw198** [phw198/OutlookGoogleCalendarSync#1883] [Outlook Online]
See #1888 for latest release updates.
------
Having extracted the development build v3.0.0.1, when running it from the command line as per the instructions (with a blank settings file), it fails with `Object reference not set to an instance of an object`.

**Comment by TheJacksonian** [comment]
Where is the best place for feedback?
I'll reply here, but happy to move if required. I get an Application unexpectedly terminated on start, 'Object reference not set to an instance of an object'.
Log attached.
[OGcalsync.log](https://github.com/user-attachments/files/15860122/OGcalsync.log)

**Reply by phw198** [reply]
I was wondering about this, and I think it's a good opportunity to test out GitHub Discussions - so I've converted this issue over to see how it goes 🙂 

So I think you missed out the keyword `call` at the start of the line invoking the OGCS executable? Also, try wrapping the `/t` parameter value in speech marks: `"Development Build"`.

**Comment by TheJacksonian** [comment]
I thought it might be due to spaces in the path, so I moved it exactly as described by you, and I was copy/pasting the call command.
![image](https://github.com/phw198/OutlookGoogleCalendarSync/assets/26889446/b6d0c012-995d-4678-a9e1-d56feaed3575)
Same issue.
[OGcalsync.log](https://github.com/user-attachments/files/15878377/OGcalsync.log)

**Reply by phw198** [reply]
OK, I've got to the bottom of that and [reissued v3.0.0.1](https://github.com/user-attachments/files/15894185/Portable_OGCS_v3.0.0.1.zip) - give it another spin...

**Comment by TheJacksonian** [comment]
Yes, I got the screen (only once yesterday, cannot reproduce) that asked for permission to automatically report the errors, and then it crashed. Now I'm getting the splash screen and the crash. Log is attached.

[OGcalsync.log](https://github.com/user-attachments/files/15906248/OGcalsync.log)

**Reply by phw198** [reply]
Did you extract the updated zip file into the same directory as before? If so, I think you have a corrupted settings.xml file in there from your first attempt, which is causing this - the file should be around 7kb in size, not 3kb or so. 

Delete that file and run OGCS again and I think it should work?

**Comment by TheJacksonian** [comment]
Success!! Deleted the settings.xml! Working through setting it up with my  dummy calendar now.

**Comment by TheJacksonian** [comment]
However, upon login, here endeth where I can assist you. There's no way I'll be getting this Admin permission. I'll discuss with some people when I'm back in the office next week, but I don't like my chances.
![image](https://github.com/phw198/OutlookGoogleCalendarSync/assets/26889446/a071bdc7-5c44-4cc3-b90c-8e5037117112)

**Reply by Zikoat** [reply]
My journey ended here as well, will report back when my admin has given me access.

**Reply by Zikoat** [reply]
I did not get any of the error messages TheJacksonian got, but i did get this warning from windows defender. It linked to this page: https://support.microsoft.com/en-us/topic/what-is-a-cloud-security-scan-75112696-7660-4450-9194-d717f72a8ad8
It doesn't seem like the security scan caused any troubles, and it took about 3 seconds before the ogcs window popped up. 

![image](https://github.com/phw198/OutlookGoogleCalendarSync/assets/10808413/a267b7f6-4a35-4168-8ee1-505fbd7c8763)

**Reply by phw198** [reply]
This is what I rather feared tbh - any company with a half decent IT department will be controlling what third party services can be used to access their data. Chances of them approving OGCS giving their data to Google may well be slim. 🫤

This is what using the locally installed fat Outlook client circumvents.

**Reply by phw198** [reply]
@Zikoat The security scan is probably due to me not code signing the build, like I do with alpha or beta releases. As long as it passed though, that's the main thing!

**Reply by TheJacksonian** [reply]
Yes, as you know I've been using the portable version for many years!!! Moving to Office360 may be the end of the road for OGCS and I. I'll chat with some people in CorpIT and see, but I may have to stick with the portable OGCS as long as they the fat Oultook on my machine.

**Reply by phw198** [reply]
@TheJacksonian I think it's only a few months off 10 years ago when I got this project going, and you were one of the early adopters then too! Fingers crossed it's not the end of the road...

I'll have a look again, but I don't even think Microsoft will let me verify the cloud app unless it's created in a corporate O365 tenant either. Let me know if _were_ verified if that would move the dial for IT allowing OGCS.

**Reply by phw198** [reply]
Interestingly, the consent screen for my corporate account looks slightly different - I guess it depend on the organisation and/or their security config...
![image](https://github.com/phw198/OutlookGoogleCalendarSync/assets/11843015/3eae2b4d-649b-4e2c-ab0d-1a0b3143bcfd)

**Reply by phw198** [reply]
Hi @TheJacksonian, @Zikoat, did either of you get a chance to discuss approving OGCS access with your IT department?

**Reply by caesay** [reply]
I use OGCS to mirror my work Outlook calendar onto a personal Google one so I can see it on my personal devices. I'm going to run into this same issue with ITOps. The "Write user and shared calendars" will likely be a blocker, since most global calendars in my company are shared in good faith, but could cause chaos if someone started mucking with them. I actually only need "Read user calendar" since I'm only syncing my personal calendar O->G. Do you think there'd be any chances we could reduce the permissions required to support this use-case?

**Reply by Zikoat** [reply]
@phw198 Response was denied from IT:
"It is not desirable to synchronize company information from a work account to a private Google account."

**Reply by faultoverload** [reply]
I think the only way to avoid the requirement to setup an app would be to use a built in browser that lets you log into OWA and then grab the auth token from that to use. You will run into issues with the refresh timers on those tokens though, not to mention having to reverse engineer the internal API used inside of OWA.

**Comment by jeme81** [comment]
Since the "new" client version may no longer require the outlook app to be installed does this open up the possibility of a linux version of this sync client?

**Reply by phw198** [reply]
Requests for MacOS version are much more common, but I've not the time to do that either, so no, very unlikely...

**Comment by mickey-bimlive** [comment]
I have installed v3.0.0.1 successfully on a Windows 10 notebook for testing, and got it working.

Now I'm trying to install it on my Windows 11 laptop. Same as on my Win 10 machine, I unpacked the zip file to C:\OCGS-dev. Then I open a command window, change directory to this directory, and execute the command `C:\OCGS-dev>call OutlookGoogleCalendarSync /t:"Development Build" /s:.\settings.xml /l:.\OGcalsync.log`. But it get the above error `Object reference not set to an instance of an object.` I deleted the setup file after this error occured and retried, but still this strange error...

My log file: [OGcalsync.log](https://github.com/user-attachments/files/16944694/OGcalsync.log)

What is wrong?

**Reply by mickey-bimlive** [reply]
Update: I got it working. The error was caused by unzipping v3.0.0.1 from https://github.com/user-attachments/files/15859398/Portable_OGCS_v3.0.0.1.zip. 

Using the zip file from https://github.com/user-attachments/files/15894185/Portable_OGCS_v3.0.0.1.zip, the error does not occur.

Both zip files are linked on different pages, and have the same name, but obviously different content. This should be corrected.

**Reply by phw198** [reply]
What are the two pages with the different links? But why are you running the original v3.0.0.1? There are several releases more recent than that...

**Reply by mickey-bimlive** [reply]
The page from which I downloaded the first 3.0.0.1 zip file is https://github.com/phw198/OutlookGoogleCalendarSync/issues/1881

After the error, I downloaded another 3.0.0.1 zip file  from the link on https://github.com/phw198/OutlookGoogleCalendarSync/discussions/1888.

Will try with the latest hotfix next.

**Reply by mickey-bimlive** [reply]
I replaced the original .exe with the one that comes with the 3.0.0.6 hotfix, but both with the 3.0.0.1 exe that comes from the link on https://github.com/phw198/OutlookGoogleCalendarSync/discussions/1888, and with the one that comes with the 3.0.0.6 hotfix, I get this error after startup:: 0x8002802B - TYPE_E_ELEMENTNOTFOUND

**Reply by mickey-bimlive** [reply]
Next I did an Office repair as advised on https://github.com/phw198/OutlookGoogleCalendarSync/wiki/FAQs---COM-Errors#0x8002802b---type_e_elementnotfound, but the 0x8002802B error persists (running 3.0.0.6)...

Log file: [OGcalsync.log](https://github.com/user-attachments/files/16955983/OGcalsync.log)

**Reply by phw198** [reply]
#1915 talks of reverting to a restore point to when Outlook wasn't broken by an upgrade. 

But if you are wanting to use OGCS v3 which doesn't need the classic Outlook client, is that even installed at all? If the classic Outlook is not meant to be installed, then I think the problem is you still have remnants of COM configured on your system:
```
2024-09-11 01:06:04,148 INFO  1 OutlookGoogleCalendarSync.Outlook.Factory [233] -  Assembly details: Microsoft.Office.Interop.Outlook, Version=15.0.0.0, Culture=neutral, PublicKeyToken=71e9bce111e9429c
2024-09-11 01:06:04,149 DEBUG  1 OutlookGoogleCalendarSync.Outlook.Factory [235] -  Assembly location: C:\WINDOWS\assembly\GAC_MSIL\Microsoft.Office.Interop.Outlook\15.0.0.0__71e9bce111e9429c\Microsoft.Office.Interop.Outlook.dll
2024-09-11 01:06:04,156 INFO  1 OutlookGoogleCalendarSync.Outlook.Calendar [1128] -  Attaching to the already running Outlook process.
```
Can you fully uninstall the classic Outlook? I guess the Outlook OGCS has detected running is the New Outlook?

**Comment by jrissman** [comment]
I know that phw198 doesn't plan to build MacOS or Linux versions, but [Wine](https://www.winehq.org/) has come a long way and now supports most modern Windows applications on both Mac and Linux platforms with few or no changes to the Windows application's code.  For ease of installation, standalone apps can be created by bundling the app with Wine using a tool like [Kegworks](https://github.com/Kegworks-App/Kegworks).

Can anyone test if the new OGCS runs under Wine on Mac (or Linux) today?  (I only have a Windows computer, so I can't test it.)  Ensuring compatibility with Wine might be the easiest way to get OGCS to run cross-platform.

**Reply by d-treegardener** [reply]
Tried to get that working for some hours now. But until now no success.
As I don't have a real windows by hand I also can't really check what should happen while starting up the first time.
In the CrossOver/Wine Logfile I get a lot of fixmes which chatgpt says ogcs is trying to open an IE, which isn't possible or at least i didn't manage to get it work.

> You’ve built a Win7 + .NET bottle without IE, hoping OGCS would open macOS’s browser. The log you posted shows the opposite: OGCS still instantiates the embedded IE WebBrowser ActiveX on startup, and Wine falls back to its stubbed mshtml/ieframe. That’s what all the noise is:
> 	•	fixme:ieframe:* / fixme:mshtml:* / fixme:jscript:* → OGCS is creating and driving the IE control (COM), but Wine’s HTML/JS engine is incomplete.
> 	•	HTMLDocumentObj_QueryInterface Unimplemented interface … and PropertyNotifySink_OnChanged unimplemented → missing COM interfaces in Wine’s mshtml; typical symptom when an embedded IE control can’t fully function.
> 
> Net: your “no IE” strategy can’t work with stock OGCS, because OGCS creates the IE control on its main form regardless. There’s no supported “use system browser” toggle in released builds, so you can’t avoid that control without either patching the code or supplying a working HTML engine in the bottle.

Digging bit deeper chatGPT recommended to change the browser helper to be less windows centric, so it could work inside of Wine. But don't know if this is really the problem ;-) as AI maybe just hallucinate about a possible solution xD

```
public static void OpenBrowser(string url) {
  try {
    var psi = new ProcessStartInfo { FileName = url, UseShellExecute = true };
    Process.Start(psi);
    return;
  } catch { /* fall through */ }

  // Wine/CrossOver fallback
  try {
    if (Environment.GetEnvironmentVariable("WINEPREFIX") != null ||
        Environment.GetEnvironmentVariable("CX_ROOT") != null) {
      Process.Start("winebrowser.exe", url); // or just "winebrowser"
      return;
    }
  } catch { /* fall through */ }

  // Windows fallback
  try {
    Process.Start("rundll32.exe", "url.dll,FileProtocolHandler " + url);
    return;
  } catch { /* fall through to registry code */ }

  // existing registry code…
}
```

**Reply by phw198** [reply]
Might be worth having a look through #2060 - @RyzenFromFire had done quite a bit of experimentation with Wine...
